pkg/cluster/kubernetes/sync.go

```diff
@@ -208,21 +207,23 @@ func (c *Cluster) getAllowedResourcesBySelector(selector string) (map[string]*ku
 		listOptions.LabelSelector = selector
 	}
 
-	_, resources, err := c.client.discoveryClient.ServerGroupsAndResources()
-	if err != nil {
-		discErr, ok := err.(*discovery.ErrGroupDiscoveryFailed)
-		if !ok {
-			return nil, err
-		}
-		for gv, e := range discErr.Groups {
-			if gv.Group == "metrics" || strings.HasSuffix(gv.Group, "metrics.k8s.io") {
-				// The Metrics API tends to be misconfigured, causing errors.
-				// We just ignore them, since it doesn't make sense to sync metrics anyways.
-				continue
-			}
-			// Tolerate empty GroupVersions due to e.g. misconfigured custom metrics
-			if e.Error() != fmt.Sprintf("Got empty response for: %v", gv) {
-				return nil, err
+	sgs, err := c.client.discoveryClient.ServerGroups()
+	if sgs == nil {
+		return nil, err
+	}
+
+	resources := []*meta_v1.APIResourceList{}
+	for i := range sgs.Groups {
+		gv := sgs.Groups[i].PreferredVersion.GroupVersion
+		// exclude the *.metrics.k8s.io resources to avoid querying the cluster metrics
+		if sgs.Groups[i].Name != "metrics.k8s.io" && !strings.HasSuffix(sgs.Groups[i].Name, ".metrics.k8s.io") {
```

Comment:
Oh, it can be _either_ of exactly metrics.k8s.io or a subdomain. Eesh. Well this is still better, I reckon -- thanks for following my suggestion up.